refl1d/mono.py

```diff
@@ -115,16 +117,19 @@ class FreeInterface(Layer):
     with slabs.
     """
 
-    name: Optional[str]
-    below: Optional[Any]
-    above: Optional[Any]
+    name: str
+    below: Scatterer
+    above: Scatterer
     thickness: Par
     interface: Par
-    dz: List[Union[float, Par]]
-    dp: List[Union[float, Par]]
+    dz: List[Par]
+    dp: List[Par]
+    magnetism: Optional[BaseMagnetism] = None
```

Comment:
Can this really be any `BaseMagnetism` object, or does it have to be specifically a `FreeMagnetismInterface`?